src/python/pants/option/custom_types.py

```diff
@@ -112,6 +112,14 @@ def dict_with_files_option(s):
   return dict_option(s)
 
 
+def shlexable_str_member_type(s):
+  """When the `type` is `list`, this can be used for the `member_type` to allow passing a shlexed
```

Comment:
This is a bit magical, as it's the only case where the value type of a list-typed option is not "list of member_type" but "flattened list of something else". 

we can make this less magical by having a `flattened_list` option type, which behaves like a list in all respects except that its value is the flattening of all member values, if they are lists themselves. 

Then we have two orthogonal pieces of behavior - shlexing a single string into a list, and flattening a list of lists into a single flat list - that together give us the effect we want, without any magic, and without any interaction between the option type and the member type.

Reply:
So basically the option would be registered as 

```
register_global('--listy-shlexable-str', type=flattened_list, member_type=shlexable, ...)
```